odoc.opam

```diff
@@ -44,8 +44,8 @@ depends: [
   "ppx_expect" {with-test}
   "bos" {with-test}
 
-  ("ocaml" {< "4.07.0" & dev} | "bisect_ppx" {dev & > "2.5.0"})
-  ("ocaml" {< "4.03.0" & dev} | "mdx" {dev})
+  ("ocaml" {< "4.07.0" & with-test} | "bisect_ppx" {with-test & > "2.5.0"})
+  ("ocaml" {< "4.03.0" & with-test} | "mdx" {with-test})
```

Comment:
`mdx` won't be co-installable when Odoc uses the new version of `odoc-parser` and we never use `bisect_ppx`.

None of `mdx` and `bisect_ppx` are required by the main test command. I suggest we drop the dependency on both.

Reply:
I don't mind. Just waiting for @jonludlam's input on your comment (we agreed on that via slack).

Reply:
The reason we don't test the mdx scripts with the default test command is because historically we couldn't guarantee it would work because of the circular dependency between mdx, odoc and odoc parser. This is no longer the case so I think we should fix this. The bisect_ppx one I'm someone less convinced about. Aspirationally I'd like to use it more :-)

Reply:
So what do you want ? 
 
> Aspirationally I'd like to use it more :-)

I would prefer if odoc devs would use `odoc` more ;-) That's what is going to make it a better tool, not manager metrics like test code coverage. Currently given all the warnings it outputs by default on any a reasonable code base, it's almost unusable. I struggle with this every day.

Reply:
Is it the warnings about not being able to find dependency modules?

Reply:
Also, the reason I want to use coverage is simply because our test suite right now is still completely insufficient to properly test the model - for a more reliable test I have to build docs using the driver, commit them to a git repo and diff them in order to make sure things haven't changed when I don't expect them to. The current process is to use this to extract a test case and drop it in, but a more systematic way would be to _at least_ try to make sure our test suite covers the majority of `xref2`. This won't be sufficient, but it'd be better than we've got now.

Reply:
> Is it the warnings about not being able to find dependency modules?

Indeed https://github.com/ocaml/odoc/issues/825